app/admin/rest.py

```diff
@@ -0,0 +1,103 @@
+from flask import Blueprint, abort, jsonify, request
+from sqlalchemy.exc import NoResultFound
+
+from app.errors import register_errors
+from app.models import (
+    ApiKey,
+    Complaint,
+    EmailBranding,
+    InboundNumber,
+    InboundSms,
+    Job,
+    LetterBranding,
+    Notification,
+    Organisation,
+    ProviderDetails,
+    Service,
+    ServiceCallbackApi,
+    ServiceContactList,
+    ServiceDataRetention,
+    ServiceEmailReplyTo,
+    ServiceInboundApi,
+    ServiceSmsSender,
+    Template,
+    TemplateFolder,
+    User,
+)
+
+admin_blueprint = Blueprint("admin", __name__)
+register_errors(admin_blueprint)
+
+FIND_BY_UUID_MODELS = {
+    "organisation": Organisation,
+    "service": Service,
+    "template": Template,
+    "notification": Notification,
+    "email_branding": EmailBranding,
+    "letter_branding": LetterBranding,
+    "user": User,
+    "provider": ProviderDetails,
+    "reply_to_email": ServiceEmailReplyTo,
+    "job": Job,
+    "service_contact_list": ServiceContactList,
+    "service_data_retention": ServiceDataRetention,
+    "service_sms_sender": ServiceSmsSender,
+    "inbound_number": InboundNumber,
+    "api_key": ApiKey,
+    "template_folder": TemplateFolder,
+    "service_inbound_api": ServiceInboundApi,
+    "service_callback_api": ServiceCallbackApi,
+    "complaint": Complaint,
+    "inbound_sms": InboundSms,
+}
+
+# The `context` here is actually information required by the admin app to build the redirect URL. This does mean
+# that this endpoint and the admin search page are fairly closely coupled. We could serialize the entire object
+# back, which might be more consistent with how other endpoints return representations of objects, but for many
+# objects we don't need any information at all, and our serialization of objects is already inconsistently between
+# `instance.serialize` and `some_schema.dump(instance)`. And not all of the serialized representations of objects
```

Comment:
hmmm, i agree with this comment (that this is probably fine, but it would be nicer if it was less coupled). if we change a route in the admin app, how long will it be before we realise this endpoint has broken?

i'm surprised that not all the objects return a relevant service_id though! which ones are those? do they seem like an intentional decision or an oversight?

if we solve those issues, if we want to decouple the admin routes from this endpoint we could always just build a dict of the serialization functions.

```py
serialize_fns = {
  Service: Service.serialize,
  Template: template_schema.dump,
  ...
}
return serialize_fns[instance.__class__](instance)
```

don't feel too strongly though, and i'm sure that approach comes with its own set of foibles

Reply:
I think Template is the first thing I hit that didn't serialize service_id (explicitly!).

I did consider having a set of serialization functions, but I think that just obscures what's actually going on - suddenly we're just implicitly _expecting_ `service_id` to be sent, but the admin app will _require_ it. It feels more fragile.

This implementation is more verbose but also more explicit about the expectations and requirements, which I think is better.